docs/concepts/performance.md

```diff
@@ -0,0 +1,182 @@
+
+In this guide, we'll introduce the factors that affect Web PubSub upstream application performance. We'll describe typical performance in different use-case scenarios. 
+
+## Term definitions
+
+*Inbound*: The incoming message to Azure Web PubSub Service.
+
+*Outbound*: The outgoing message from Azure Web PubSub Service.
+
+*Bandwidth*: The total size of all messages in 1 second.
+
+## Overview
+
+Azure Web PubSub Service defines seven Standard tiers for different performance capacities. This
+guide answers the following questions:
+
+- What is the typical Azure Web PubSub Service performance for each tier?
+
+- Does Azure Web PubSub Service meet my requirements for message throughput (for example, sending 100,000 messages per second)?
+
+- For my specific scenario, which tier is suitable for me? Or how can I select the proper tier?
+
+To answer these questions, this guide first gives a high-level explanation of the factors that affect performance. It then illustrates the maximum inbound and outbound messages for every tier for typical use cases: **Send to groups through Web PubSub subprotocol**, **upstream** and **rest api** .
+
+This guide can't cover all scenarios (and different use cases, message sizes, message sending patterns, and so on). But it provides some basic information to understand the performance limitation.
+
+## Performance insight
+
+This section describes the performance evaluation methodologies, and then lists all factors that affect performance. In the end, it provides methods to help you evaluate performance requirements.
+
+### Methodology
+
+*Throughput* and *latency* are two typical aspects of performance checking. For Azure Web PubSub Service, each SKU tier has its own throughput throttling policy. The policy defines *the maximum allowed throughput (inbound and outbound bandwidth)* as the maximum achieved throughput when 99 percent of messages have latency that's less than 1 second.
+
+### Performance factors
+
+Theoretically, Azure Web PubSub Service capacity is limited by computation resources: CPU, memory, and network. For example, more connections to Azure Web PubSub Service cause the service to use more memory. For larger message traffic (for example, every message is larger than 2,048 bytes), Azure Web PubSub Service needs to spend more CPU cycles to process traffic.
+
+The message routing cost also limits performance. Azure Web PubSub Service plays a role as a message router, which routes the message among a set of clients. A different scenario or API requires a different routing policy. 
+
+For **echo**, the client sends a message to itself, and the routing destination is also itself. This pattern has the lowest routing cost. But for **broadcast**, **send to group**, and **send to connection**, Azure Web PubSub Service needs to look up the target connections through the internal distributed data structure. This extra processing uses more CPU, memory, and network bandwidth. As a result, performance is slower.
```

Comment:
> sends a message to itself

feels a little confusing, a client is unable to send a message to itself. How about: "sends a message to the upstream, and upstream echos the message back to the client"?